File: sjf_llama.py
def SJF(jobs, index):
    job_list = [(i, job) for i, job in enumerate(jobs)]
    
    sorted_jobs = sorted(job_list, key=lambda x: (x[1], x[0]))
    
    total_time = 0
    target_index = index
    
    for i, (job_idx, job_cc) in enumerate(sorted_jobs):
        if job_idx == target_index:
            return total_time + job_cc
        
        total_time += job_cc
    
    return -1

SJF([100], 0)
SJF([3,10,20,1,2], 0)
SJF([3,10,20,1,2], 1)
SJF([3,10,20,1,2,3], 5)
SJF([3,10,20,1,2,10,10], 5)

File: sjf_llama.rs
fn sjf(jobs: Vec<i32>, index: usize) -> i32 {
    let mut jobs = jobs
       .into_iter()
       .enumerate()
       .map(|(i, job)| (i, job))
       .collect::<Vec<_>>();

    let mut current_cc = 0;
    let mut finished = vec![];

    while jobs.len() > 0 {
        jobs.sort_by_key(|&(_, job)| job);

        let (i, mut job) = jobs.remove(0);

        while job > 0 {
            job -= 1;

            current_cc += 1;

            if job == 0 {
                finished.push(i);
            }
        }
    }

    finished
       .into_iter()
       .enumerate()
       .find(|&(_, i)| i == index)
       .map(|(cc, _)| cc as i32 + 1)
       .unwrap()
}

sjf(&[100], 0)
sjf(&[3,10,20,1,2], 0)
sjf(&[3,10,20,1,2], 1)
sjf(&[3,10,20,1,2,3], 5)
sjf(&[3,10,20,1,2,10,10], 5)